requirements.txt

```diff
@@ -1,6 +1,16 @@
 # remember to also update unittest_requirements.txt when updating this file
+aiohttp>=3.8.1,<=3.8.3
+cryptography>=38.0.1<=38.0.1
+Django~=3.2.15  # LTS version, switch only if we have a compelling reason to
+djangorestframework>=3.13.1,<=3.14.0
+django-filter>=22.1,<=22.1
+django-import-export>=2.9,<=2.9.0
+drf-access-policy>=1.1.2,<=1.1.2
+drf-spectacular==0.24.2  # We monkeypatch this so we need a very narrow requirement string
```

Comment:
Do you directly import this?

Anything pulp_container doesn't directly import doesn't need to be declared. Perhaps you do directly import this though....

Reply:
yeah we do
```
[ipanova@puffy pulp_container]$ git grep spectac
.ci/scripts/schema.py:This script modifies drf-spectacular schema validation to accept slashes and curly brackets.
.ci/scripts/schema.py:from drf_spectacular.validation import JSON_SCHEMA_SPEC_PATH
pulp_container/app/viewsets.py:from drf_spectacular.utils import extend_schema
[ipanova@puffy pulp_container]$ git grep extend_schema
pulp_container/app/viewsets.py:from drf_spectacular.utils import extend_schema
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
pulp_container/app/viewsets.py:    @extend_schema(
[ipanova@puffy pulp_container]$ 
```

Reply:
I would expect us to have a _very_ broad range here. Pinning to the same version as pulpcore is calling for desaster. Also i think the monkey-patching does not apply here.

Reply:
I agree, I think this can be very broad and it needs to be or this strategy won't work well.

Reply:
As an aside, we need pulpcore to stop pinning.

Reply:
As long as we do the monkey patching there is no stopping.

Reply:
Nothing without badly breaking the client bindings.

Reply:
@bmbouter @mdellweg what range do you expect to be used here?

Reply:
I would still call drf-spectacular part of the pulpcore plugin interface. But I do not intend to reroll the whole discussion from the beginning.
How about `drf-spectacular>=0.24,<0.25`?

Reply:
I'd be ok with either drf-spectacular>=0.24,<0.25 or not declaring it at all.